### DISCUSSION TITLE
RTK Query - Handle result global?

### DISCUSSION THREAD

**Discussion by wuifdesign** [reduxjs/redux-toolkit#1392] [Q&A]
Is there a way to handle the result of every endpoint inside an `createApi`.

For example: 

- I want to dispatch an `authenticated: false` and show a login overlay whenever any of the endpoints returns a 401.
- If the result is a JSON and includes an `notifications` key, i want to display these notifications as messages.

It seams like this is only possible if i add this to each implementation of any of these endpoints.

**Comment by phryneas** [comment]
For something like that, you would wrap your baseQuery in a custom function.

You can find an example here: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#automatic-re-authorization-by-extending-fetchbasequery

**Reply by wuifdesign** [reply]
thx :)
